canvas_sdk/v1/data/base.py

```diff
@@ -35,13 +36,51 @@ def find(self, value_set: Type["ValueSet"]) -> models.QuerySet:
 
         Condition.objects.find(MorbidObesity).find(AnaphylacticReactionToCommonBakersYeast)
         """
-        values_dict = value_set.values
-        uri_codes = [
-            (i[1], values_dict[i[0]])
-            for i in value_set.CODE_SYSTEM_MAPPING.items()
-            if i[0] in values_dict
-        ]
         q_filter = Q()
-        for system, codes in uri_codes:
-            q_filter |= Q(codings__system=system, codings__code__in=codes)
+        for system, codes in self.codings(value_set):
+            q_filter |= Q(**self.q_kwargs(system, codes))
         return self.filter(q_filter).distinct()
+
+    @staticmethod
+    def codings(value_set: Type["ValueSet"]) -> tuple[tuple[str, set[str]]]:
+        """Provide a sequence of tuples where each tuple is a code system URL and a set of codes."""
+        values_dict = value_set.values
+        return cast(
+            tuple[tuple[str, set[str]]],
+            tuple(
+                (i[1], values_dict[i[0]])
+                for i in value_set.CODE_SYSTEM_MAPPING.items()
+                if i[0] in values_dict
+            ),
+        )
+
+    @staticmethod
+    def q_kwargs(system: str, codes: Container[str]) -> dict[str, str | Container[str]]:
+        """
+        This method can be overridden if a Q object with different filtering options is needed.
+        """
+        return {"codings__system": system, "codings__code__in": codes}
```

Comment:
@jrwils Let me know what you think about this. I did a little refactoring to accommodate Questionnaire. Two changes:

* Provide the codings via a method that can be overridden. See the subclass `ValueSetLookupByNameQuerySet` for why this was needed.
* Provide the Q object kwargs via a method that can be overridden. See the subclass `QuestionnaireValueSetLookupQuerySet` for why this was needed.

Reply:
This all makes sense to me 👍